### PR TITLE
Ability to display only Lure Pokestops (AHAAAAAAA#1954)

### PR DESCRIPTION
**Pull request by georgo**
## Description
Added switcher to allow display only lured Pokestops

There is one thing I'm not sure about. It's loading of pokestops in raw_data, please verify it.

## How Has This Been Tested?
Tested on Chrome and iOS 9.x (iPhone 6)

## Screenshots (if appropriate):
screenshot_20160725_122119

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by sp90** [comment]
Looks really great

**Comment by invisiblek** [comment]
I'm not sure I'm on board with the UX here. Does disabling lured pokestops, but leaving regular pokestops show only non-lured pokestops? Should it?

Would spawning a second option upon turning on pokestops to hide non-lured pokestops make more sense?

(Just trying to drum up some conversation on this, it feels off to me)

**Comment by Sp4rkR4t** [comment]
Personally I feel changing the Pokestops options to s drop down with Off, Only Lured & All as the option would be the better way to go with this.

**Comment by AHAAAAAAA** [comment]
I believe a checkbox under Show Pokestops titled 'Only Lured' would work best. It should be an extension of Show Pokestops, not its own toggle, having 2 toggles is confusing.

**Comment by Sp4rkR4t** [comment]
Checkbox would be fine, I just thought since there are already drop downs in the filter menu it would make sense to keep to that.

**Comment by W1ndst0rm** [comment]
I think I agree with @Sp4rkR4t on this. Turning it into a drop down keeps the interface from getting overwhelmed and cluttered with toggles and checkboxes. Just my 2 cents.

**Comment by georgo** [comment]
Let me change it for ya.

**Comment by georgo** [comment]
Code is now ready to review. Switcher is now dropdown with 2 options, special function to display only lured pokestops is now removed.

screenshot_20160726_084914

File: static/map.js [hunk 1]
```diff
@@ -748,16 +753,26 @@ function processPokestops(i, item) {
...
     }
+    if (Store.get('showLuredPokestops') && !item.lure_expiration) {
+        if (map_data.pokestops[item.pokestop_id] && map_data.pokestops[item.pokestop_id].marker) {
+            map_data.pokestops[item.pokestop_id].marker.setMap(null);
+        }
+        return true;
+    }
     if (map_data.pokestops[item.pokestop_id] == null) { // add marker to map and item to dict
         // add marker to map and item to dict
-        if (item.marker) item.marker.setMap(null);
```
Comment (invisiblek) [inline static/map.js]: unecessary change for this patch

File: static/map.js [hunk 2]
```diff
@@ -134,6 +134,10 @@ var StoreOptions = {
         default: true,
         type: StoreTypes.Boolean
     },
+    showLuredPokestops: {
```
Comment (invisiblek) [inline static/map.js]: should this be showOnlyLuredPokestops ?

File: static/map.js [hunk 3]
```diff
@@ -314,6 +318,7 @@ function initSidebar() {
     $('#gyms-switch').prop('checked', Store.get('showGyms'));
     $('#pokemon-switch').prop('checked', Store.get('showPokemon'));
     $('#pokestops-switch').prop('checked', Store.get('showPokestops'));
+    $('#lured-pokestops-switch').val(Store.get('showLuredPokestops'));
```
Comment (invisiblek) [inline static/map.js]: lured-pokestops-only-switch ?

File: static/map.js [hunk 4]
```diff
@@ -748,16 +753,26 @@ function processPokestops(i, item) {
...
+            item.marker.setMap(null);
+        }
         item.marker = setupPokestopMarker(item);
         map_data.pokestops[item.pokestop_id] = item;
     }
     else {
         item2 = map_data.pokestops[item.pokestop_id];
         if (!!item.lure_expiration != !!item2.lure_expiration || item.active_pokemon_id != item2.active_pokemon_id) {
-            item.marker.setMap(null);
+            if (item.marker) {
```
Comment (invisiblek) [inline static/map.js]: unecessary change for this patch

Comment (georgo) [inline static/map.js]: Removed patch.

File: static/map.js (hunk 3 above)
Comment (georgo) [inline static/map.js]: Good idea. Applied.

File: static/map.js (hunk 2 above)
Comment (georgo) [inline static/map.js]: To preserve convention, I kept showLuredPokestopsOnly.

File: static/map.js (hunk 1 above)
Comment (georgo) [inline static/map.js]: Patch removed.

**Comment by georgo** [comment]
Rebased and applied required changes.

**Comment by invisiblek** [comment]
We probably want to hide the drop down if the toggle is off, otherwise this is looking good to me.

**Comment by georgo** [comment]
That shouldn't be problem. Let me do that tomorrow morning ;)

**Comment by CarlTaylor1989** [comment]
+1 tested and works, make the change @invisiblek suggested and good to merge.

**Comment by AHAAAAAAA** [comment]
+1 too, as soon as we get those changes!

**Comment by georgo** [comment]
Display of switcher added. I tried to keep animation, hope you will like it ;)

**Comment by MonkeyLeeT** [comment]
Wait wut animation? @georgo

**Pull request merged**